File: python/lesson 1/examples.py
class Examples:
    def sum(self, nums: List[int]) -> int:
        result = 0
        for i in range(len(nums)):
            result += nums[i]
        return result

    def buildMatrix(self, rows: int, cols: int) -> List[List[int]]:
        result = [None] * rows
        for row in range(rows):
            result[row] = [None] * cols
            for col in range(cols):
                result[row][col] = (row + 1) * (col + 1)                
        return result

    def sumRecursive(self, nums: List[int]) -> int:
        return self.sumRecursiveImpl(nums, 0)

    def sumRecursiveImpl(self, nums: List[int], current: int) -> int:
        if current >= len(nums): return 0
        return nums[current] + self.sumRecursiveImpl(nums, current + 1)

File: ruby/lesson 1/examples.rb
def sum(nums)
    result = 0
    (0..nums.length - 1).each { |i| result += nums[i] }
    result
end

def count_zero_sum_not_optimized(nums)
    result = 0
    (0..nums.length - 1).each do |i| 
        (i + 1..nums.length - 1).each do |j|
            result += 1 if nums[i] + nums[j] == 0
        end        
    end
    result
end

def sum_recursive(nums)
    sum_recursive_impl(nums, 0);
end

def sum_recursive_impl(nums, current)
     return 0 if current >= nums.length

    nums[current] + sum_recursive_impl(nums, current + 1)
end
